sql/updates/post-update.sql

```diff
@@ -0,0 +1,5 @@
+set timescaledb.altering_extension = true;
```

Comment:
This need not be a GUC. You can just `SET LOCAL foo=true` a variable and read it with ` GetConfigOption(")` in the C code. LOCAL means the setting will be cleared at the end of the transaction, which is what we want. Ideally, the `SET LOCAL` line should be injected prior to the post-update.sql.

I suggest using a variable `timescaledb.update_script_stage=post`, which we can also use in the future to have multiple stages (e.g., "pre", "post").